### PR TITLE
feat: Add support for path-based file lookup

### PR DESCRIPTION
**Pull request by cmellwig**
Allow disambiguating files with the same name in different directories by using path/filename syntax in links. 

For example:

    :myapi:`src/bsp/drv/adc.c`
    :myapi:`src/hal/peripheral/adc.c`

Backward compatibility with existing links is maintained.

### CODE REVIEW

File: sphinxcontrib/doxylink/doxylink.py [hunk 1]
```diff
@@ -385,8 +397,17 @@ def find_doxygen_link(name, rawtext, text, lineno, inliner, options={}, content=
             warning_messages.append('Could not find match for `%s` because tag file not found' % part)
             return [nodes.inline(title, title)], []
 
+        # Handle path:filename format
+        target_path = None
+        target_name = part
+        if ':' in part:
+            target_path, target_name = part.split(':', 1)
```
Comment (JasperCraeghs) [inline sphinxcontrib/doxylink/doxylink.py:404]: Shouldn't we use [`rsplit`](https://docs.python.org/3/library/stdtypes.html#str.rsplit) instead? 

Here, you use a colon as the separator, but the example in the PR description uses a forward slash. What am I missing? Where would the colon character come from?

Comment (JasperCraeghs) [reply sphinxcontrib/doxylink/doxylink.py:404]: If I understand correctly, the use of a colon is optional and you have invented the path:filename format. I don't see how it adds any value, who would use it like this:
```
:myapi:`src/bsp/drv/:adc.c`
```
I am happy to merge the rest of this PR after documenting the feature and adding a unit test for it.

**Review by JasperCraeghs — CHANGES_REQUESTED** [review]
## Review

I dug into the colon-based `path:filename` syntax and found a correctness issue that I think needs fixing before this can merge.

`find_doxygen_link()`'s `if ':' in part:` check fires on *any* namespace-qualified symbol — which is doxylink's primary use case. E.g. `PolyVox::Array` splits into `('PolyVox', ':Array')` and gets reconstructed as `PolyVox:Array`, which won't match any real tag-file entry. I verified this directly with a few namespaced examples; they all break.

There's also a path-joining issue: `join()` (used for combining path + filename) is bare string concatenation with no separator. Depending on whether Doxygen happens to emit a trailing slash on `<path>`, either the old unqualified syntax or the new `path:filename` syntax ends up broken — I reproduced both failure modes with synthetic tag files matching the repo's existing fixtures.

To get this to a mergeable state, I'd suggest:
- Picking a delimiter that can't appear in a C++ symbol (colon collides with `::`), or scoping the new syntax so it only kicks in when there's no `::` present
- Fixing the join so exactly one separator is always inserted between path and filename (e.g. `os.path.normpath(os.path.join(...))`)
- Adding tests that go through `create_role`/`find_doxygen_link` (not just `SymbolMap`) — none currently exercise this new code path, including namespaced symbols and tag files with/without a trailing slash on `<path>`
- A README/docs mention of the new syntax, since right now it's undiscoverable

Happy to look again once these are addressed — the feature itself (disambiguating same-named files) is a real and useful thing to support.

*(Review assisted by Claude Code on behalf of the maintainer.)*

**Comment by JasperCraeghs** [comment]
Closing this out — it's been about 16 months without a follow-up on the review feedback, and the underlying feature (disambiguating same-named files) is worth having, so I've reimplemented it in #78 using a different approach that avoids the `::`/`:` collision this PR's syntax had: instead of a new `path:filename` delimiter, each file's tag-file `<path>` is joined onto its name with `/` (which can never appear in a C++ symbol), reusing the existing suffix-matching. `:myapi:\`adc.c\`` still works unchanged, and `:myapi:\`src/hal/peripheral/adc.c\`` disambiguates.

Thanks for surfacing the idea and the initial implementation — sorry it took so long to land in some form. Happy to revisit if you have thoughts on the approach in #78.
